src/components/external/DisputeManagement/components/DefendDisputeFlow/DefendDisputeFileUpload.tsx

```diff
@@ -108,6 +110,34 @@ export const DefendDisputeFileUpload = () => {
         return docs;
     }, [applicableDocuments, i18n]);
 
+    const areRequiredDocsUploaded = useMemo(() => {
+        if (!defendDisputePayload) return false;
+        const uploaded = new Map(defendDisputePayload.entries());
+
+        const areAllRequiredPresent = requiredDocuments.every(d => uploaded.has(d) && !!uploaded.get(d));
+        if (oneOrMoreDocuments.length === 0) {
+            return areAllRequiredPresent;
+        }
+
+        const isAtLeastOneOptionalPresent = oneOrMoreDocuments.some(d => uploaded.has(d.id) && !!uploaded.get(d.id));
+        return areAllRequiredPresent && isAtLeastOneOptionalPresent;
```

Comment:
- From the logic you have here, you don't need a new `Map` object for this, you can perform all the queries directly on the `FormData` object.
- When checking for uploaded files, you can skip the `has()` check, since you are following-up with a `get()` query. It's best to be sure a file was uploaded, rather than just checking for the presence of a value.

```suggestion
        const allRequiredDocumentsPresent = requiredDocuments.every(d => defendDisputePayload.get(d) instanceof File);
        if (oneOrMoreDocuments.length === 0) return allRequiredDocumentsPresent;

        const atLeastOneOptionalDocumentPresent = oneOrMoreDocuments.some(d => defendDisputePayload.get(d.id) instanceof File);
        return allRequiredDocumentsPresent && atLeastOneOptionalDocumentPresent;
```